src/main/java/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -323,8 +318,13 @@ public void setQueryContext(QueryContext queryContext) {
     private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance sourceInstance,
                                             TreeReference workingRef, Vector<TreeReference> refs,
                                             boolean includeTemplates) {
-
-        int depth = workingRef.size();
+        int depth;
+        try {
+            depth = workingRef.size();
+        } catch (NullPointerException e) {
+            throw new RuntimeException("Invalid instance definition encountered while evaluating " + sourceRef.toString() +
```

Comment:
Another minor style quibble: when we do catch-and-throw I think it's good to get into the habit of including the original throwable as the context (second arg to the RuntimeException). That would, for instance, let someone reviewing the stack trace identify the weird side-case where the NPE was thrown from inside `size()`

Reply:
yeah totally agree. [The commit](https://github.com/dimagi/commcare-core/pull/926/commits/5d6e4d3d1e78043e272ab56f35ac08784d14ec63) did make the exact correction earlier though it's no longer relevant.